src/granite/query/assemblers/base.cr

```diff
@@ -38,19 +38,29 @@ module Granite::Query::Assembler
     def where
       return @where if @where
 
-      clauses = @query.where_fields.map do |expression|
+      first = true
+      clauses = [] of String
+
+      @query.where_fields.each do |expression|
         add_aggregate_field expression[:field]
 
+        if first
+          first = false
+          clauses << "WHERE"
+        else
+          clauses << expression[:join].to_s.upcase
+        end
+
         if expression[:value].nil?
-          "#{expression[:field]} IS NULL"
+          clauses << "#{expression[:field]} IS NULL"
         else
-          "#{expression[:field]} #{sql_operator(expression[:operator])} #{add_parameter expression[:value]}"
+          clauses << "#{expression[:field]} #{sql_operator(expression[:operator])} #{add_parameter expression[:value]}"
         end
       end
 
       return nil if clauses.none?
 
-      @where = "WHERE #{clauses.join " AND "}"
+      @where = clauses.join(" ")
     end
 
```

Comment:
There is probably a better, less empirical way to do this.  Please use github's  new `suggestion` feature to provide a solution.